src/lsst/cmservice/web_app/static/js/script-grid-utils.js

```diff
@@ -0,0 +1,192 @@
+// list of statuses to reset to if current status is failed or rejected
+const resetFromRejected = {
+    WAITING: 0,
+    READY: 1,
+    PREPARED: 2
+};
+// list of statuses to reset to if current status is reviewable
+const resetFromReviewable = {
+    ACCEPTED: 5,
+    REJECTED: -3,
+};
+
+const updateStatus = () => {
+    const rowIndex = parseInt(document.querySelector("#rowIndex").innerText);
+    const newStatus = parseInt(document.querySelector("#targetStatus").value);
+    const fullname = String(document.querySelector("#scriptFullname").innerText)
+    const scriptId = scriptGridApi.getRowNode(rowIndex).data.id;
+
+    let url;
+    let resetJson = null;
+
+    // accept script
+    if(newStatus === 5){
+        url = `/cm-service/v1/script/action/${scriptId}/accept`;
+    }
+    // reject script
+    else if(newStatus === -3){
+        url = `/cm-service/v1/script/action/${scriptId}/reject`;
+    }
+    // reset script
+    else {
+        let resetObj = {
+            fullname: fullname,
+            status: newStatus,
+        };
+
+        resetJson = JSON.stringify(resetObj);
+        url = reset_script_endpoint;
+    }
+
+    // send request
+    let xhr = new XMLHttpRequest();
+    xhr.open('POST', url, true);
+    xhr.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');
+    xhr.onload = function () {
+        if (xhr.status === 201) {
+            // update script status in grid
+            scriptGridApi
+                .getRowNode(rowIndex)
+                .setDataValue("status", document.querySelector("#targetStatus").selectedOptions[0].text);
+            // refresh grid cells on client side to disable/enable reset button according to new status
+            scriptGridApi.refreshCells({
+                force: true,
+                suppressFlash: false,
+              });
+            resetScriptModal.close();
+        } else {
+            resetScriptModal.close();
+            errorModal.showModal();
+            errorMessage.innerText = JSON.parse(xhr.responseText).detail;
+        }
+    };
+    xhr.onerror = function () {
+        console.error("Error:", xhr.status, xhr.responseText);
+        resetScriptModal.close();
+        errorModal.showModal();
+        errorMessage.innerText = xhr.responseText;
+    };
+    xhr.send(resetJson);
+
+};
+
+const readScriptLog = (log_url) => {
+
+    const xhr = new XMLHttpRequest();
+
+    const url = "/web_app/read-script-log";
+
+    xhr.onreadystatechange = function () {
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            if (xhr.status === 200) {
+                const response = JSON.parse(xhr.responseText);
+                // show file content from response in the textarea
+                document.querySelector('#scriptLogContent').value = response.content;
+                scriptLogModal.showModal();
+            } else {
+                console.error("Error:", xhr.status, xhr.responseText);
```

Comment:
Logging errors to the console outside of dev is probably not a great idea. Do we need to log this? Usually in production you avoid any kind of console logging.

Reply:
Nope. I'll remove console logs